### PR TITLE
Restrict jupyterHubRole

### PR DESCRIPTION
**Pull request by wmuizelaar**
Based on the contents of [https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/master/jupyterhub/templates/hub/rbac.yaml](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/master/jupyterhub/templates/hub/rbac.yaml)

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/895)

### CODE REVIEW

**Comment by wmuizelaar** [comment]
I signed it!

—
Met vriendelijke groet/Kind regards,

Wietse Muizelaar | Lead System Engineer | bol.com
Bezoek: Papendorpseweg 100, 3528 BJ, Utrecht | Post: Postbus 8080, 3503 RB Utrecht
T: +31 (0)88 7123915 | M: +31 (0)6 54618736 | E-mail: wmuizelaar@bol.com

**Comment by pdmack** [comment]
/ok-to-test

@wmuizelaar, can you tidy up the PR title and description please?

**Comment by pdmack** [comment]
@wmuizelaar there's a test that covers that yaml. 

`kubeflow/core/tests/jupyterhub_test.jsonnet`

Can you update that too and run `scripts/autoformat_jsonnet.sh` on it?

```
Assertion failed. {"apiVersion": "rbac.authorization.k8s.io/v1beta1", "kind": "Role", "metadata": {"name": "jupyter-role", "namespace": "test-kf-001"}, "rules": [{"apiGroups": ["*"], "resources": ["pods", "persistentvolumeclaims"], "verbs": ["get", "watch", "list", "create", "delete"]}, {"apiGroups": ["*"], "resources": ["events"], "verbs": ["get", "watch", "list"]}]} != {"apiVersion": "rbac.authorization.k8s.io/v1beta1", "kind": "Role", "metadata": {"name": "jupyter-role", "namespace": "test-kf-001"}, "rules": [{"apiGroups": ["*"], "resources": ["*"], "verbs": ["*"]}]}
```

**Comment by wmuizelaar** [comment]
@pdmack thanks for the pointer! I did so, squashed the commits, and it looks like the tests are rerun now?

File: kubeflow/core/jupyterhub.libsonnet [hunk 1]
```diff
@@ -219,13 +219,30 @@ c.RemoteUserAuthenticator.header_name = 'x-goog-authenticated-user-email'",
           apiGroups: [
             "*",
           ],
-          // TODO(jlewi): This is very permissive so we may want to lock this down.
           resources: [
```
Comment (jlewi) [inline kubeflow/core/jupyterhub.libsonnet]: Can you add a comment with a link to the file you have in the PR. This will help keep it in sync over time.

**Comment by jlewi** [comment]
/approve

@pdmack You are as much an OWNER of all things JupyterHub as I'm so feel free to approve PRs without me.

**Comment by pdmack** [comment]
@jlewi ack
You had the initial TODO so I deferred.

**Pull request merged**